### PR TITLE
Rewrite `consists_of` option as dynamic and refuse collection mode

### PR DESCRIPTION
**Pull request by afuno**

### CODE REVIEW

File: spec/examples/usual/dynamic_options/consists_of/example13_spec.rb [hunk 1]
```diff
@@ -68,7 +68,7 @@
             expect { perform }.to(
               raise_error(
                 ApplicationService::Exceptions::Input,
-                "[Usual::Collection::Example13] Required element in input collection `ids` is missing"
+                "Input `ids` must be a collection of `String`"
```
Comment (afuno) [inline spec/examples/usual/dynamic_options/consists_of/example13_spec.rb:71]: This change is correct. Technically the attribute value is present. The fact that there is an incorrect value inside the collection is the responsibility of the dynamic option.

File: spec/examples/usual/dynamic_options/consists_of/example13_spec.rb [hunk 2]
```diff
@@ -87,7 +87,7 @@
...
```
Comment (afuno) [inline spec/examples/usual/dynamic_options/consists_of/example13_spec.rb:90]: This change is correct. Technically the attribute value is present. The fact that there is an incorrect value inside the collection is the responsibility of the dynamic option.

File: spec/examples/usual/dynamic_options/consists_of/example13_spec.rb [hunk 3]
```diff
@@ -194,7 +192,7 @@
...
```
Comment (afuno) [inline spec/examples/usual/dynamic_options/consists_of/example13_spec.rb:195]: This change is correct. Technically the attribute value is present. The fact that there is an incorrect value inside the collection is the responsibility of the dynamic option.

File: spec/examples/usual/dynamic_options/consists_of/example14_spec.rb [hunk 4]
```diff
@@ -68,7 +68,7 @@
             expect { perform }.to(
               raise_error(
                 ApplicationService::Exceptions::Input,
-                "[Usual::Collection::Example14] Required element in input collection `ids` is missing"
+                "Input `ids` must be a collection of `String`"
```
Comment (afuno) [inline spec/examples/usual/dynamic_options/consists_of/example14_spec.rb:71]: This change is correct. Technically the attribute value is present. The fact that there is an incorrect value inside the collection is the responsibility of the dynamic option.

File: spec/examples/usual/dynamic_options/consists_of/example14_spec.rb [hunk 5]
```diff
@@ -87,7 +87,7 @@
...
```
Comment (afuno) [inline spec/examples/usual/dynamic_options/consists_of/example14_spec.rb:90]: This change is correct. Technically the attribute value is present. The fact that there is an incorrect value inside the collection is the responsibility of the dynamic option.

File: spec/examples/usual/dynamic_options/consists_of/example14_spec.rb [hunk 6]
```diff
@@ -175,7 +173,7 @@
...
```
Comment (afuno) [inline spec/examples/usual/dynamic_options/consists_of/example14_spec.rb:176]: This change is correct. Technically the attribute value is present. The fact that there is an incorrect value inside the collection is the responsibility of the dynamic option.

File: spec/examples/usual/dynamic_options/consists_of/example14_spec.rb [hunk 7]
```diff
@@ -194,7 +192,7 @@
...
```
Comment (afuno) [inline spec/examples/usual/dynamic_options/consists_of/example14_spec.rb:195]: This change is correct. Technically the attribute value is present. The fact that there is an incorrect value inside the collection is the responsibility of the dynamic option.

File: spec/examples/usual/dynamic_options/consists_of/example5_spec.rb [hunk 8]
```diff
@@ -87,7 +87,7 @@
             expect { perform }.to(
               raise_error(
                 ApplicationService::Exceptions::Input,
-                "[Usual::Collection::Example5] Required element in input collection `ids` is missing"
+                "Input `ids` must be an array of `String`"
```
Comment (afuno) [inline spec/examples/usual/dynamic_options/consists_of/example5_spec.rb:90]: This change is correct. Technically the attribute value is present. The fact that there is an incorrect value inside the collection is the responsibility of the dynamic option.

File: spec/examples/usual/dynamic_options/consists_of/example5_spec.rb [hunk 9]
```diff
@@ -194,7 +192,7 @@
...
```
Comment (afuno) [inline spec/examples/usual/dynamic_options/consists_of/example5_spec.rb:195]: This change is correct. Technically the attribute value is present. The fact that there is an incorrect value inside the collection is the responsibility of the dynamic option.

File: spec/examples/usual/dynamic_options/consists_of/example5_spec.rb [hunk 10]
```diff
@@ -175,7 +173,7 @@
...
```
Comment (afuno) [inline spec/examples/usual/dynamic_options/consists_of/example5_spec.rb:176]: This change is correct. Technically the attribute value is present. The fact that there is an incorrect value inside the collection is the responsibility of the dynamic option.

File: spec/examples/usual/dynamic_options/consists_of/example6_spec.rb [hunk 11]
```diff
@@ -87,7 +87,7 @@
             expect { perform }.to(
               raise_error(
                 ApplicationService::Exceptions::Input,
-                "[Usual::Collection::Example6] Required element in input collection `ids` is missing"
+                "Input `ids` must be an array of `String`"
```
Comment (afuno) [inline spec/examples/usual/dynamic_options/consists_of/example6_spec.rb:90]: This change is correct. Technically the attribute value is present. The fact that there is an incorrect value inside the collection is the responsibility of the dynamic option.

File: spec/examples/usual/dynamic_options/consists_of/example6_spec.rb [hunk 12]
```diff
@@ -68,7 +68,7 @@
...
```
Comment (afuno) [inline spec/examples/usual/dynamic_options/consists_of/example6_spec.rb:71]: This change is correct. Technically the attribute value is present. The fact that there is an incorrect value inside the collection is the responsibility of the dynamic option.

File: spec/examples/usual/dynamic_options/consists_of/example6_spec.rb [hunk 13]
```diff
@@ -175,7 +173,7 @@
...
```
Comment (afuno) [inline spec/examples/usual/dynamic_options/consists_of/example6_spec.rb:176]: This change is correct. Technically the attribute value is present. The fact that there is an incorrect value inside the collection is the responsibility of the dynamic option.

File: spec/examples/wrong/dynamic_options/consists_of/example1_spec.rb [hunk 14]
```diff
@@ -30,8 +30,8 @@
           expect { perform }.to(
             raise_error(
               ApplicationService::Exceptions::Internal,
-              "[Wrong::Collection::Example1] Wrong internal attribute collection " \
-              "type `ids`, expected `Set`, got `Array`"
+              "[Wrong::DynamicOptions::ConsistsOf::Example1] Wrong type of internal attribute `ids`, " \
```
Comment (afuno) [inline spec/examples/wrong/dynamic_options/consists_of/example1_spec.rb:33]: This change is correct.

File: spec/examples/wrong/dynamic_options/consists_of/example1_spec.rb [hunk 15]
```diff
@@ -76,8 +78,8 @@
...
```
Comment (afuno) [inline spec/examples/wrong/dynamic_options/consists_of/example1_spec.rb:81]: This change is correct.

File: spec/examples/wrong/dynamic_options/consists_of/example1_spec.rb [hunk 16]
```diff
@@ -76,8 +78,8 @@
...
+              "expected `Set`, got `Array`"
```
Comment (afuno) [inline spec/examples/wrong/dynamic_options/consists_of/example1_spec.rb:82]: This change is correct.

File: spec/examples/wrong/dynamic_options/consists_of/example1_spec.rb [hunk 17]
```diff
@@ -30,8 +30,8 @@
...
```
Comment (afuno) [inline spec/examples/wrong/dynamic_options/consists_of/example1_spec.rb:34]: This change is correct.

File: lib/servactory/tool_kit/dynamic_options/consists_of.rb [hunk 18]
```diff
@@ -0,0 +1,92 @@
...
+
+module Servactory
+  module ToolKit
+    module DynamicOptions
+      class ConsistsOf < Must
+        COLLECTION_CLASS_NAMES = [Array, Set].freeze
+        private_constant :COLLECTION_CLASS_NAMES
+
+        def self.setup(option_name = :consists_of)
+          new(option_name, :type, false).must(:consists_of)
```
Comment (nflm-codex) [inline lib/servactory/tool_kit/dynamic_options/consists_of.rb:11]: а что за `:type, false` ?
не вижу этих атрибутов в базовом классе https://github.com/servactory/servactory/blob/main/lib/servactory/tool_kit/dynamic_options/must.rb#L7

Comment (anton-co) [reply lib/servactory/tool_kit/dynamic_options/consists_of.rb:11]: Ты смотришь старый код. Посмотри здесь ниже, это следующий файл.

Comment (nflm-codex) [reply lib/servactory/tool_kit/dynamic_options/consists_of.rb:11]: отмена, увидел ниже

**Review by nflm-codex — APPROVED** [review]

**Pull request merged**